src/pages/ZNS/ZNS.tsx

```diff
@@ -193,6 +194,11 @@ const ZNS: React.FC<ZNSProps> = ({ domain, version }) => {
 		if (triedEagerConnect)
 			addNotification(active ? 'Wallet connected.' : 'Wallet disconnected.');
 		// eslint-disable-next-line react-hooks/exhaustive-deps
+		
+	if(disconnectState === "true" && active!){
+		deactivate();
```

Comment:
This code doesn't work if I disconnect my wallet, then clear my cache/local storage for the app. The dApp will then go and reconnect. This solution works, but not in all cases.